### DISCUSSION TITLE
Cannot insert promotion in SAN format to the Board

### DISCUSSION THREAD

**Discussion by mikkoville** [tors42/chariot#9] [Q&A] [answered]
Hi,

I was trying out this project and I could not insert the following pgn to the board due to it having an illegal move.

`"e4 c6 d4 d5 Nc3 Nd7 exd5 cxd5 Bb5 e6 Nf3 Nf6 Ne5 Be7 h4 a6 Ba4 b5 Bb3 Nxe5 dxe5 Nd7 f4 Bb7 Bd2 Bxh4+ Rxh4 Qxh4+ Kf1 d4 Ne2 Qh1+ Kf2 Qxg2+ Ke1 h5 Nxd4 h4 Qe2 Qg1+ Qf1 Qg3+ Kd1 h3 f5 h2 fxe6 h1=Q"`

The last move of promotion gives an error:
`Invalid move: Invalid[info=move: c1=Q fen: 6k1/r5p1/P3Np1p/3P4/4P3/4b1P1/1rp4P/R4R1K b - - 0 33]`

I created a new board with `Board.fromStandardPosition()` and passed the above moves to it with` Board play(String var1);`

Thanks!

**Comment by tors42** [comment, marked as answer]
Oh indeed, I could reproduce the problem with the help of your data - thanks for the report!

I've released chariot-0.0.68,
which intends to fix this problem.

**Comment by tors42** [comment]
Thanks for marking the answer!

I must admit,
the chariot.util.Board implementation is not something which I have deeply reflected upon nor something I "feel proud" of -
it is more some kind of minimum effort implementation just to have some basic chess rule functionality in place.
 
Chariot is mainly supposed to be a Lichess API library,
but I got a bit carried away,
and added the Board functionality in this library,
as my interest in chess grew stronger while learning more about the game.

I really appreciate that you found and reported the problem so it could be fixed,
but I fear that there might exist more problems,
and I also fear that this chess library isn't the most user friendly chess library API available...

So if you are looking for a general Java chess library,
maybe you would be better off using some other chess library out there...

Thanks again!

**Reply by mikkoville** [reply]
Thanks! Yeah I was using it only for some testing and found this thing. Thanks for the heads up also 👍
